js/src/scatter.js

```diff
@@ -263,10 +287,15 @@ var ScatterView = widgets.WidgetView.extend( {
         // add atrributes to the geometry, this makes the available to the shader
         current.add_attributes(instanced_geo)
         previous.add_attributes(instanced_geo, '_previous')
-
-	    this.mesh = new THREE.Mesh(instanced_geo, this.material );
+        var material;
```

Comment:
This section below here swaps out the new material. I wasn't sure what material_rgb does, or where it gets used... Not sure if this is the best way to do this swap.

Reply:
material_rgb should write rgb color corresponding to the normalised (0, 0, 0) (1, 1, 1) which is used for the volume rendering to know where opaque objects are, so where to stop the ray.

Reply:
Yes, it is when volume rendering and scatters are mixed in one plot, for instance:
```python
fig = ipv.figure()
cube = ipv.datasets.aquariusA2.fetch().data
ipv.volshow(cube, extent=[[-4, 4]]*3)
g = ipv.examples.gaussian(N=100);
```
Maybe crank up the opacity for the first level to see the scatter points fade away. If you now change the geo to a sprint, it will not render properly.
For debugging, you can set: `fig.show = 'Back' # or 'Front' or 'Volume'`